tools/snp.sh

```diff
@@ -1107,16 +1152,102 @@ attest_guest() {
 
   # Remove any special characters and print the value
   snpguest_report_measurement=$(echo ${snpguest_report_measurement} | sed $'s/[^[:print:]\t]//g')
+  echo -e "Measurement from SNP Attestation Report: ${snpguest_report_measurement}\n"
+
+	  # Compare the expected measurement to the guest report measurement
+	  [[ "${expected_measurement}" == "${snpguest_report_measurement}" ]] \
+		&& echo -e "The expected measurement matches the snp guest report measurement!" \
+		|| { >&2 echo -e "FAIL: measurements do not match"; return 1; }
+else 
+ 	 Verify_measurement
+fi
+}
+setup_svsm_guest_attestation() {
+
+   ssh_guest_command "
+   
+    # Update and install necessary packages
+    echo 'Updating package list...'
+    sudo apt-get update
+
+    echo 'Installing necessary packages...'
+    sudo apt-get install -y git build-essential libtss2-dev tpm2-tools
+
+    echo 'Installing Rust...'
+    source ""${HOME}"/.cargo/env" 2>/dev/null || true
+
+    curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs -sSf | sh -s -- -y
+    source ""${HOME}"/.cargo/env" 2>/dev/null
+    "    
+
+  # Check if 'snpguest' directory exists and remove it if it does
+  ssh_guest_command "
+  if [ -d \"snpguest\" ]; then
+     echo 'Directory snpguest exists. Removing it...'
+     rm -rf snpguest
+  else
+     echo 'Directory snpguest does not exist.'
+  fi
+"
+
+  # Clone the repository and build the project
+  ssh_guest_command "git clone https://github.com/virtee/snpguest.git"
+  ssh_guest_command "cd snpguest  &&  source \$HOME/.cargo/env &&  cargo build --release"
+}
+
+Verify_measurement(){
+
+  # Define the working directory and the output file
+  OUTPUT_FILE="$SETUP_WORKING_DIR/make_output.txt"
+  # Define the firmware file path
+  FW_FILE="$SETUP_WORKING_DIR/AMDSEV/ovmf/Build/OvmfX64/DEBUG_GCC5/FV/OVMF.fd"
+
+  cd "$SETUP_WORKING_DIR"/AMDSEV/svsm
+
+  # Run the make command and save the output to a file
+  FW_FILE="$FW_FILE" make &> "$OUTPUT_FILE"
+
+  # Initialize the variable for Launch Digest
+  launch_digest=""
+
+  # Read the output file line by line to find the Launch Digest
+  while IFS= read -r line; do
+    if [[ "$line" == *"Launch Digest:"* ]]; then
+      launch_digest=$(echo "$line" | awk '{print $3}')
+      break
+    fi
+  done < "$OUTPUT_FILE"
```

Comment:
I think all of this logic could be reduced.

You're running the make command, which I'm assuming at one point it uses igvmmeasure to calculate the measurement, but it also creates a lot of extra stuff that you put into your $OUTPUT_FILE. and then you read through this output file to look for the expected measurement. 

Can you not simply call igvmmeasure instead of running the whole make command? Idk what else is being created with this make command? IS there anything else you need in the $OUTPUT_FILE that is generated?

Reply:
Yes, I will check. The igvmmeasure tool is an integral part of the SVSM compilation process. I will see if I can run it individually. I will update verify measurement process.